packages/dds/tree/src/test/shared-tree/fuzzEditGenerator.ts

```diff
@@ -0,0 +1,331 @@
+                            nodeIndex = fieldNodes !== 0 ? random.integer(0, fieldNodes - 1) : 0;
+                            cursor.free();
+                            return { parent: parentPath, field: nodeField, index: nodeIndex };
+                        } else {
+                            nodeField = testerKey;
+                            nodeIndex = 0;
+                        }
+                        break;
+                    }
+                } else {
+                    cursor.free();
+                    return {
+                        parent: parentPath,
+                        field: nodeField,
+                        index: 0,
+                    };
+                }
+                break;
+            case "firstField":
+                if (cursor.firstField()) {
+                    currentMove = random.pick(moves.field);
+                    fieldNodes = cursor.getFieldLength();
+                    nodeField = cursor.getFieldKey();
+                } else {
+                    cursor.free();
+                    return { parent: parentPath, field: testerKey, index: 0 };
+                }
+                break;
+
+            case "nextField":
+                if (cursor.nextField()) {
+                    currentMove = random.pick(moves.field);
+                    fieldNodes = cursor.getFieldLength();
+                    nodeField = cursor.getFieldKey();
+                } else {
+                    currentMove = "stop";
+                    nodeField = testerKey;
+                    nodeIndex = 0;
+                }
+                break;
+            default:
+                fail(`Unexpected move ${currentMove}`);
+        }
+    }
+    cursor.free();
+    return { parent: parentPath, field: nodeField, index: nodeIndex };
+}
+
+export function getExistingRandomNodePosition(
+    tree: ISharedTree,
+    random: IRandom,
+): UpPath | undefined {
+    const cursor = tree.forest.allocateCursor();
+    moveToDetachedField(tree.forest, cursor);
+    const firstNode = cursor.firstNode();
+    assert(firstNode, "tree must contain at least one node");
+    const firstPath = cursor.getPath();
+    assert(firstPath !== undefined, "firstPath must be defined");
+    let path: UpPath = firstPath;
+    const firstField = cursor.firstField();
+    if (!firstField) {
+        // no fields, return the rootnode
+        cursor.free();
+        return path;
+    }
+    let fieldNodes: number = cursor.getFieldLength();
+    let nodeIndex: number = 0;
+
+    let currentMove = random.pick(moves.field);
+    assert(cursor.mode === CursorLocationType.Fields);
+
+    while (currentMove !== "stop") {
+        switch (currentMove) {
+            case "enterNode":
+                if (fieldNodes > 0) {
+                    nodeIndex = random.integer(0, fieldNodes - 1);
+                    cursor.enterNode(nodeIndex);
+                    const currentPath = cursor.getPath();
```

Comment:
`getPath` should never return undefined here. I recommend either asserting that it doesn't or just doing `cursor.getPath()!;` (in which case you'll have to quiet the lint warning)